packages/pigeon/lib/swift_generator.dart

```diff
@@ -668,19 +683,15 @@ String _camelCase(String text) {
 }
 
 String _castForceUnwrap(String value, TypeDeclaration type, Root root) {
-  final String forceUnwrap = type.isNullable ? '' : '!';
-  final String castUnwrap = type.isNullable ? '?' : '';
   if (isEnum(root, type)) {
-    final String nullableConditionPrefix =
-        type.isNullable ? '$value == nil ? nil : ' : '';
-    return '$nullableConditionPrefix${_swiftTypeForDartType(type)}(rawValue: $value as! Int)$forceUnwrap';
+    return '${_swiftTypeForDartType(type)}(rawValue: $value as! Int)!';
```

Comment:
This is wrong for nullable types without the nullableConditionPrefix you removed; if `value` is `nil` or `NSNull` this cast is invalid. Are we not using this codepath?

Reply:
When you say "this code path" do you mean the way it's currently called, rather than something in the method? Because if so, that seems very dangerous; it would be easy to violate that later without even knowing it wasn't intended to work.

If we can't make something self-contained work here, can we instead move this elsewhere, and `assert` that `type` isn't an enum here so this isn't a danger?

Reply:
This line of code is all that's needed for non-nullable enums.

Nullable enums need extra code to check for nil before creating the enum from the int that is provided.

I had merged these lines into a single ternary before, but that was causing some other issues. I think I've resolved those now though, so I'll give it another try

Reply:
I fixed this in a really jank way, going to send up a better version tomorrow

Reply:
This still has the issue that if you call it with a nullable enum it will happily output crashing code.

Reply:
That's true, I can nest this method inside the other to make it uncallable. It is never used except in the context of _writeDecodeCasting

Reply:
or this function could just be removed entirely, since it is not really needed any longer.